content/6_Advanced/LineContainer.mdx

```diff
@@ -30,7 +30,191 @@ frequency: 1
+
+To visualize this, we shall let $H = A,B,C,D,E$. This is the set of half-planes that are currently present in the intersection. For the points, we can define them as $P = p,q,r,s$ which are intersection points of adjacent half-planes from $H$. This could be represented as such below:
+
+<iframe
+	src="https://iq.opengenus.org/content/images/2022/02/halfplane2.PNG"
+	width="651px"
+	height="491px"
+	frameborder="0"
+/>
+
+Say we want to intersect it with a new half-plane $F$; it may make some existing half-planes redundant.
+
+<iframe
+	src="https://iq.opengenus.org/content/images/2022/02/halfplane3.PNG"
+	width="735px"
+	height="492px"
+	frameborder="0"
+/>
+
+We can see $F$ here represented in orange. It makes the half-planes $A$ and $E$ redundant. Therefore, we can remove them from the front and back of the queue and add $F$ at the end. Then, we obtain a new intersection between $H = B,C,D,F$ with $P = q,r,t,u$. This can be shown below:
+
+<iframe
+	src="https://iq.opengenus.org/content/images/2022/02/halfplane4.PNG"
+	width="707px"
+	height="416px"
+	frameborder="0"
+/>
+
+This is the main principle of the algorithm. However, there are a few special cases we should cover. In the case of parallel half-planes, there are 2 possible scenarios, either two that are parallel with the same direction or different directions. We should approach parallel lines with different directions the same as if we have to increase the size of the bounding box, since there will have to be at least one of the half-planes within the bounding box in between the two since they are sorted by angle. This means that the only case we have to deal with is where there are two half-planes with the same angle. To do this is straightforward: simply keep the leftmost half-plane and erase the rest since the others will eventually become redundant anyway.
+
+So, as an overview of the algorithm:
+* Sort the set of half-planes by angle; it takes $\mathcal{O}(NlogN)$ time.
+* Iterate over the set. For each, perform incrementation and pop either from the front or back of the double-ended queue as necessary; it takes linear time since every half-plane can only be added or removed once.
+* Once this has been completed, the convex polygon that results from the intersection can be obtained by computing the intersection points of adjacent half-planes in the double-ended queue. It will take linear time. This results in our final complexity as $\{O}(NlogN)$. In a special case where the half-planes are pre-sorted, this algorithm could be completed in $\{O}(N)$. However, for the majority of cases, we will be looking at $\{O}(NlogN)$, most of which is coming from the sorting by angle.
+
+#### Implementation
+
+```cpp
+#include <climits>
+#include <deque>
+#include <vector>
+#include <algorithm>
+
+const long double eps = 1e-9;
+const int inf = INT32_MAX;
+
+struct Point {
+    long double x, y;
+    explicit Point(long double x = 0, long double y = 0) : x(x), y(y) {}
```

Comment:
hey man you know the precommit bot is changing things to tabs for a reason right

Reply:
so, what do u want me to change?

Reply:
no like i just find it interesting how pre commit changed everything to tabs and yet you always change it back to spaces